src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs

```diff
@@ -0,0 +1,177 @@
+                    if (mappingResult != null)
+                    {
+                        if (mappingResult.Symbol is INamedTypeSymbol namedTypeSymbol)
+                        {
+                            // Find its baseTypes and subTypes.
+                            await AddInheritanceMemberItemsForNamedTypeAsync(
+                                mappingResult.Project.Solution,
+                                sourceText,
+                                memberDeclarationNode,
+                                namedTypeSymbol, builder, cancellationToken).ConfigureAwait(false);
+                        }
+
+                        if (mappingResult.Symbol.IsOrdinaryMethod() || mappingResult.Symbol is IEventSymbol or IPropertySymbol)
+                        {
+                            // Find the implementing/implemented/overridden/overriding members
+                            await AddInheritanceMemberItemsForTypeMembersAsync(
+                                mappingResult.Project.Solution,
+                                sourceText,
+                                memberDeclarationNode,
+                                mappingResult.Symbol,
+                                builder,
+                                cancellationToken).ConfigureAwait(false);
+                        }
+                    }
+                }
+            }
+
+            return builder.ToImmutable();
+        }
+
+        private async Task AddInheritanceMemberItemsForNamedTypeAsync(
+            Solution solution,
+            SourceText sourceText,
+            SyntaxNode declarationNode,
+            INamedTypeSymbol memberSymbol,
+            ArrayBuilder<InheritanceMemberItem> builder,
+            CancellationToken cancellationToken)
+        {
+            // Get all base types.
+            var allBaseTypes = BaseTypeFinder.FindBaseTypesAndInterfaces(memberSymbol);
+
+            // Filter out
+            // 1. System.Object. (otherwise margin would be shown for all classes)
+            // 2. System.ValueType. (otherwise margin would be shown for all structs)
+            // 3. System.Enum. (otherwise margin would be shown for all enum)
+            var baseTypes = allBaseTypes
+                .WhereAsArray(symbol => !IsUnwantedBaseType((ITypeSymbol)symbol));
+
+            // Get all derived types
+            var derivedTypes = await GetDerivedTypesAndImplementationsAsync(
+                solution,
+                memberSymbol,
+                cancellationToken).ConfigureAwait(false);
+
+            if (baseTypes.Any() || derivedTypes.Any())
+            {
+                var lineNumber = GetIdentifierLineNumber(sourceText, declarationNode);
+                var item = await CreateInheritanceMemberInfoAsync(
+                    solution,
+                    memberSymbol,
+                    lineNumber,
+                    baseSymbols: baseTypes,
+                    derivedTypesSymbols: derivedTypes.CastArray<ISymbol>(),
```

Comment:
the inconsistency with tehse names bugs me.  bastSymbols-bastTypes.  baseSymbols-derivedTypesSymbols :)

Reply:
Change all to 'Symbols"